c++/src/kj/compat/tls.c++

```diff
@@ -419,20 +419,26 @@ private:
 
 class TlsConnectionReceiver final: public ConnectionReceiver, public TaskSet::ErrorHandler {
 public:
-  TlsConnectionReceiver(TlsContext &tls, Own<ConnectionReceiver> inner)
+  TlsConnectionReceiver(
+      TlsContext &tls, Own<ConnectionReceiver> inner, kj::Maybe<ErrorHandler&> errorHandler)
       : tls(tls), inner(kj::mv(inner)),
         acceptLoopTask(acceptLoop().eagerlyEvaluate([this](Exception &&e) {
           onAcceptFailure(kj::mv(e));
         })),
-        tasks(*this) {}
+        tasks(errorHandler.orDefault(*this)) {}
 
   void taskFailed(Exception&& e) override {
-    // TODO(someday): SSL connection failures may be a fact of normal operation but they may also
-    // be important diagnostic information. We should allow for an error handler to be passed in so
-    // that network issues that affect TLS can be more discoverable from the server side.
-    if (e.getType() != Exception::Type::DISCONNECTED) {
-      KJ_LOG(ERROR, "error accepting tls connection", kj::mv(e));
-    }
+    // Note that if an ErrorHandler was provided to the ctor, this will not be used.
+    switch (e.getType()) {
+      case Exception::Type::DISCONNECTED:
+      case Exception::Type::OVERLOADED: {
+        return;
+      } break;
```

Comment:
Nitpick: This bracing style for switch cases isn't used anywhere else in KJ.

* We normally wouldn't add braces if there's no variables defined in the scope.
* When we do have braces, the `break` goes inside the braces.
* We don't normally write `break` if the case already ended in a `return`.

I understand that you may have different preferences here but consistency is important. Using an inconsistent style may lead other readers to wonder why the code here looks different and whether they are missing something important.

Note that we compile with `-Werror=implicit-fallthrough` which will catch any case fallthrough that isn't explicitly annotated.